lib/all-badge-examples.js

```diff
@@ -2219,6 +2219,9 @@ function findCategory(wantedCategory) {
 
 function loadExamples() {
   loadServiceClasses().forEach(ServiceClass => {
+    if (ServiceClass.category === 'debug') {
+      return;
```

Comment:
I'm not sure, but can we replace it with this?
```js
if (!ServiceClass.examples) {
```

Reply:
That feels like a good check to make, but perhaps it should fail a test or issue a warning, unless category is `debug`.

Reply:
Normally if we wanted to create a badge but not add it to the examples page (for some reason), it would have been sufficient to define:

```js
static get examples() {
  return [];
}
```

The reason I have specifically added a check here is because I've defined this badge with a category 'debug' which doesn't exist in the examples. If we remove that check, the next line `const category = findCategory(ServiceClass.category);` throws `Error: Unknown category debug referenced in Time`. If I had (wrongly IMO) defined this badge as being in some other category like 'downloads', just returning `[]` from `examples()` would be fine and I wouldn't have added this check. We want some way to ensure we don't call `findCategory()` on a category that doesn't exist.

It is probably also useful if:

* There is some way to explicitly declare that particular badges should be excluded from the examples (we also do this with `flip`, for example)
* loadExamples() throws an exception if `example()` is not implemented at all on some class, as in most cases that probably represents an error or oversight.

I don't have particularly strong opinions on the structure or convention we choose to to do that. Thoughts? (Note that `[]` is 'truthy' in javascript.)

> Perhaps it should check an env var like ENABLE_DEBUG_SERVICES

Remember we're assembling examples for the front-end here. I'm not sure it is too important that it is configurable whether we show this debug badge on the examples page (presumably you'd want that on in local/dev if anywhere)?

Reply:
Gotcha! It didn't register that this was in `loadExamples`.

Could you move the check into the block right below?

i.e.
```js
if (category === undefined) {
  if (ServiceClass.category === 'debug') {
    return;
  } else {
    throw …
  }
}
```